service/interpreter/workflowImpl.go

```diff
@@ -239,11 +240,12 @@ func executeState(
 			DataObjects:      persistenceManager.LoadDataObjects(state.StateOptions),
 		},
 	}).Get(ctx, &startResponse)
-	if err != nil {
-		return nil, convertStateApiActivityError(provider, err)
+
+	if errorFromActivity != nil && (state.StateOptions == nil || state.StateOptions.StartApiFailurePolicy == nil || state.StateOptions.GetStartApiFailurePolicy() == iwfidl.FAIL_WORKFLOW_ON_START_API_FAILURE) {
```

Comment:
nit: move `state.StateOptions == nil || state.StateOptions.StartApiFailurePolicy == nil || state.StateOptions.GetStartApiFailurePolicy() == iwfidl.FAIL_WORKFLOW_ON_START_API_FAILURE` to a helper method like `shouldProceedOnStartApiError(state)`